wkcuber/converter.py

```diff
@@ -0,0 +1,283 @@
+                # already inside the mag folder
+                add_or_update_if_present(layers, "", "")
+                dataset_names.add("dataset")
+            elif len(split_path) == 5:
+                # only the mag folder is given, therefore the layer path is empty
+                add_or_update_if_present(layers, "", split_path[0])
+                dataset_names.add("dataset")
+            elif len(split_path) == 6:
+                # additionally the layer folder is given, that should indicate a single layer as well
+                add_or_update_if_present(layers, split_path[0], split_path[1])
+                dataset_names.add("dataset")
+            else:
+                # also a dataset folder is given
+                layer_path = sep.join(split_path[0:-5])
+                if starts_with_root:
+                    layer_path = "/" + layer_path
+                add_or_update_if_present(layers, layer_path, split_path[-5])
+                dataset_names.add(split_path[-7])
+
+        assert (
+            len(dataset_names) == 1
+        ), "More than one dataset found. Stopping conversion..."
+        assert len(layers) > 0, "No layers found. Stopping conversion..."
+
+        layer_names_and_mags = dict()
+
+        for layer, mags in layers.items():
+            layer_names_and_mags[layer] = natsorted(list(mags))[0]
+
+        return dataset_names.pop(), layer_names_and_mags
+
+
+class ImageStackConverter(Converter):
+    def __init__(self) -> None:
+        self.source_files: List[str] = []
+        self.args: Namespace = Namespace()
+
+    def accepts_input(self, source_path: str) -> bool:
+        source_files = get_source_files(source_path, image_reader.readers.keys(), True)
+
+        if len(source_files) == 0:
+            return False
+
+        _, ext = path.splitext(source_files[0])
+
+        assert all(
+            map(lambda p: path.splitext(p)[1] == ext, source_files)
+        ), "Not all image files are of the same type"
+
+        self.source_files = source_files
+
+        return True
+
+    def convert_input(self, args: Namespace) -> None:
+        print("Converting image stack")
+
+        # add missing config attributes with defaults
+        put_default_if_not_present(args, "target_mag", 1)
+        put_default_if_not_present(args, "wkw_file_len", 32)
+        put_default_if_not_present(args, "interpolation_mode", "default")
+        put_default_if_not_present(args, "start_z", 0)
+        put_default_if_not_present(args, "jobs", cpu_count())
+        put_default_if_not_present(args, "distribution_strategy", "multiprocessing")
+        put_default_if_not_present(args, "job_resources", None)
+        put_default_if_not_present(args, "pad", False)
+        put_default_if_not_present(args, "max_mag", 32)
+        put_default_if_not_present(args, "no_compress", False)
+        put_default_if_not_present(args, "isotropic", False)
+        put_default_if_not_present(args, "verbose", True)
```

Comment:
Since they are only defined inside the argument parser, I am not sure how to handle this in a more DRY fashion. The first idea that comes to my mind, would be to set the default of every value to `None` in the argument parser and have a similar validator check this after argument parsing. Then we could have a unified default args dict inside each class. Do you think this is a good idea?